### ISSUE TITLE
unnecessary_nested_if_linter() false positive

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#2443]
Found in data.table, e.g. https://github.com/Rdatatable/data.table/blob/3bd4fd16395b015fcb8f72db54948f7cc5562710/R/bmerge.R#L134-L137

```r
lint(linters = unnecessary_nesting_linter(), text = "
if (isTRUE(getOption('datatable.use.index'))) {
  xo = getindex(x, names(x)[xcols])
  if (verbose && !is.null(xo)) catf('on= matches existing index, using index\n')
}
")
# <text>:4:3: warning: [unnecessary_nesting_linter] Don't use nested `if` statements, where a single `if` with the combined conditional expression will do. For example, instead of `if (x) { if (y) { ... }}`, use `if (x && y) { ... }`.
#   if (verbose && !is.null(xo)) catf('on= matches existing index, using index
#   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I suspect the lack of `{` on the inner statement is throwing off the logic.
